src/Items.py

```diff
@@ -25,7 +25,14 @@
 
 # add sequential generated ids to the lists
 for key, val in enumerate(item_table):
-    item_table[key]["id"] = count
+    if "id" in item_table[key]:
+        item_id = item_table[key]["id"]
+        if item_id > count:
+            item_table[key]["id"] = item_id
+        else:
+            raise ValueError(f"{item_table[key]['name']} has an invalid ID. ID must be at least {count + 1}")
+    else:
+        item_table[key]["id"] = count
```

Comment:
I think this breaks in a situation where world creators only set some of the item ids. Like, if I have these items:

- Item 1
- Item 2, id = 4
- Item 3
- Item 4

Then id 4 will be set for 2 items.

We might want to set count to the custom id when we encounter it. Then the increment starts counting from there.

Reply:
> We might want to set count to the custom id when we encounter it. Then the increment starts counting from there.  

either that or we keep track of the custom ids somewhere and skip over them when counting up

Reply:
Updating count is absolutely the intended behaviour, thanks for catching that